### PR TITLE
Grant Application for Hydrapad

### PR DESCRIPTION
**Pull request by hydrapad**
#### Project Abstract

> Dear Grants Committee,
We are excited to submit our application to support the development & integration of Hydrapad on Polkadot, a decentralized presale launchpad platform designed to facilitate seamless, transparent, and secure fundraising for web3 startups.
>
> If your application is a follow-up to a previous grant, please mention which one in the first line of the abstract and include a link to previous pull requests if applicable.

#### Grant [level](https://github.com/w3f/Grants-Program#level_slider-levels)

- [ ] **Level 1**:  Up to $10,000, 2 approvals
- [x] **Level 2**:  Up to $30,000, 3 approvals
- [ ] **Level 3**:  Unlimited, 5 approvals (for >$100k: Web3 Foundation Council approval)

#### Application Checklist

- [x] The [application template](https://github.com/w3f/Grants-Program/blob/master/applications/application-template.md) has been copied and aptly renamed (`project_name.md`).
- [x] I have read the [application guidelines](https://github.com/w3f/Grants-Program/blob/master/docs/Support%20Docs/grant_guidelines_per_category.md).
- [x] Payment details have been provided (Polkadot AssetHub (USDC & DOT) address in the application and bank details via email, if applicable).
- [x] I understand that an agreed upon percentage of each milestone will be paid in vested DOT, to the Polkadot address listed in the application.
- [x] I am aware that, in order to receive a grant, I (and the entity I represent) have to successfully complete a KYC/KYB check.
- [x] The software delivered for this grant will be released under an open-source license specified in the application.
- [x] The initial PR contains only one commit (squash and force-push if needed).
- [x] The grant will only be announced once the first milestone [has been accepted](https://github.com/w3f/Grant-Milestone-Delivery#process) (see the [announcement guidelines](https://github.com/w3f/Grants-Program/blob/master/docs/Support%20Docs/announcement-guidelines.md)).
- [ ] I prefer the discussion of this application to take place in a private Element/Matrix channel. My username is: `@_______:matrix.org` (change the homeserver if you use a different one)

### CODE REVIEW

**Review by semuelle — CHANGES_REQUESTED** [review]
Thanks for the application, @hydrapad. Most of our team is out of office during the holidays, so it will take at least another week before someone can give a thorough response. 

In the meantime, can you elaborate on what you mean by "will be built on Substrate" if you already have a working version on EVM? Also, can you share some insight into your userbase and successes on Ethereum? What made you consider adding Polkadot?

Please also add technical details, such as which smart contract language you will be using and deliverables for each milestone. Please also review our guidelines and FAQ, so that you are familiar with what activities we do and do not fund:

- FAQ: https://grants.web3.foundation/docs/faq#what-activitiespositions-do-you-fund
- Guidelines: https://grants.web3.foundation/docs/Introduction/intro

**Comment by hydrapad** [comment]
Hello @semuelle good to connect and happy holidays. We have not launched our product yet. However we did create contracts for EVM, which are now being audited, Apart from that we also would like to have Polkadot integrated on our platform as there are no Launchpad on Polkadot ecosystem which provides bonded curve mechanism with live trading. So it's an area where we can provide our services. 

**Here are few more reasons below:**

- Polkadot’s unique multi-chain framework offers an unparalleled advantage in terms of interoperability. With its relay chain and parachains, Polkadot enables seamless communication between different blockchains. For a bonded curve launchpad, where assets and liquidity need to be bridged across various ecosystems, Polkadot’s cross-chain capabilities are a natural fit. This ensures we can facilitate broader participation and integration with different blockchain communities, making our platform more accessible and scalable.

- Substrate's modular framework allows us to customize the blockchain layer according to our needs, especially as we design a bonded curve mechanism for our launchpad. Substrate’s flexibility in governance, consensus mechanisms, and runtime modules makes it easy to tailor the platform's features while maintaining high performance. This adaptability means we can innovate rapidly and efficiently.

- Polkadot’s parachains can scale independently, enabling our project to grow alongside the broader ecosystem. By utilizing Polkadot’s infrastructure, we can ensure that our launchpad can handle increased transaction volumes as our user base and features expand, without compromising performance or decentralization.

**Note:** Similar bonded curve presale launchpad exists on Solana called Pump.fun, We want to bring similar concept to Polkadot ecosystem, along side with EVM chains. And our platform will have a lot more features & security mechanism to prevent rugpulls / unwanted activities. Plus free token creation + free Presale creation with just one click. Our aim is to lift high entry barrier for any projects to come and launch on Polkadot and provide sustainable ecosystem.

**Technical Details: Smart Contract Language and Integration on Polkadot**

- For the smart contract component of the launchpad, we will use Ink!, which is a Rust-based programming language designed for writing WebAssembly (Wasm) smart contracts on Substrate-based chains. Ink! is highly optimized for Substrate and enables us to deploy smart contracts directly on Polkadot parachains, ensuring both security and performance. It's well-suited for our use case because it allows us to build flexible complex logic, while also taking advantage of the native Polkadot parachain infrastructure for scalability and cross-chain functionalities.

- By integrating with parachains, we can ensure cross-chain liquidity and asset transfers, which are crucial for a launchpad that needs to interact with multiple ecosystems (e.g., different token standards, DEXs, or other decentralized projects).

- For cross-chain communication, we plan to utilize XCMP (Cross-Chain Message Passing) to facilitate trustless interactions between parachains or the Polkadot relay chain.

- By using Wasm smart contracts (via Ink!) and Substrate’s native architecture, our launchpad will benefit from the security and performance optimizations of Polkadot and Substrate. Ink! contracts are executed in the Polkadot runtime environment, ensuring that they inherit the security guarantees and consensus mechanisms of Polkadot and its parachains.

- The substrate runtime also ensures that our smart contracts are highly efficient and can be executed in a gas-efficient manner, reducing unnecessary costs for users interacting with the launchpad.

- Polkadot’s ecosystem provides future-proof integration with other blockchain ecosystems, and through its bridges (e.g., with Ethereum, Bitcoin, or Cosmos), our launchpad could potentially support token sales, liquidity pools, and other activities across various blockchain networks. This is a critical advantage of building on Polkadot, as it aligns with our goal of creating a highly interoperable launchpad.

In conclusion, Polkadot is home to a growing and diverse range of projects, which means that our launchpad can leverage its ecosystem to attract new users, developers, and projects looking for a launchpad solution with integrated liquidity. Polkadot’s strategic vision aligns with our mission to create a sustainable, decentralized platform that supports the broader Web3 movement. Please feel free to ask any questions you have or perhaps we can jump in a live call to explain in more detailed manner. Thanks

**Comment by hydrapad** [comment]
We will update the deliverables for each milestone soon. Thank you.

**Comment by hydrapad** [comment]
I have read and hereby sign the Contributor License Agreement.

File: applications/Hydrapad.md [hunk 1]
```diff
@@ -0,0 +1,282 @@
...
+
+2. Matthew Lopez is a blockchain strategist skilled in ecosystem growth and community adoption. He specializes in creating sustainable models for scalable blockchain solutions. With extensive experience in education, hackathons, and financial inclusion, he bridges emerging technologies with underserved region
+
+#### Team Code Repos
+
+- https://github.com/orgs/HYDRAPAD-LAUNCHPAD/
+
+Please also provide the GitHub accounts of all team members. If they contain no activity, references to projects hosted elsewhere or live are also fine.
+
+- https://github.com/orgs/HYDRAPAD-LAUNCHPAD/people/wonderwomancode - Angela Steffens
```
Comment (semuelle) [inline applications/Hydrapad.md:164]: This link 404s

**Review by semuelle — CHANGES_REQUESTED** [review]
Thanks for the updates and sorry for the wait, @hydrapad. 

The issue with the CLA is because @chainemesis still needs to sign. Other than that, I still have a few questions:

- pump.fun has gotten into trouble in the past for rugpulls. What are these "features & security mechanism to prevent rugpulls / unwanted activities" that you mention? 
- Are you planning to KYC users? Otherwise, how are you planning to ensure that you are complying with all laws and regulations that your users might be subject to?
- Does anyone on your team have experience with Polkadot SDK and/or ink!? Otherwise, five weeks for initial platform integration seems highly ambitious.
- Your comments on the advantages of ink! and Polkadot SDK are a little vague. Could you provide an architecture diagram or sequence diagrams for how you imagine - especially cross-chain transactions - would work? 
- What parachains are you planning to deploy on?

**Comment by hydrapad** [comment]
@semuelle good to hear from you. 

**Features & security mechanisms to prevent rug pulls/unwanted activities**
- **Bonded Curve Mechanism:** Our presale platform utilizes a bonded curve, which ensures that token prices are algorithmically determined by supply and demand. This prevents manipulation of token prices, providing transparent and fair pricing for all participants.

- **1. Escrow and Vesting:** Funds raised during the presale are held in an escrow account until a project reaches certain predefined milestones or until a lockup period has expired. Additionally, tokens are vested in a way that restricts the immediate withdrawal or selling of large amounts, helping prevent early investors from dumping tokens and causing market disruption. 
For Example: For Bonded Curve Fair Launch  Presale: user create a token + Presale deployment in one application. 100% of the token created stays within the contract, doesn't get sent to project owner's address. 10% of the token gets vested automatically for 6 months. 90% of the token gets reserved for the presale. If project owners wants to have their own token before the vesting period is over, they will have to buy just like any other investors would. Once the presale is over, the liquidity pairs get created on the designated Dex and LP burned. Project owner gets 5% of the DOT raised from the presale to cover operational expenses and community management. After 6 months project owner will receive 10% of the the vested token. So there is very little room for rugpulls.

- **Security Audits:** All smart contracts are subjected to rigorous third-party audits from reputable blockchain security firms. We also adopt best practices in smart contract development, including the use of formal verification tools and secure development frameworks.

- **Community Reporting:** Our platform integrates community-driven tools to report any suspicious activity, with a transparent dispute resolution mechanism in place. This helps us quickly identify and respond to potential risks in the ecosystem.

By combining these advanced features with strong community oversight, we are able to significantly mitigate the risk of rug pulls and other unwanted activities on the platform.

**2. Regarding KYC users and how are we planning to ensure that we are complying with all laws and regulations that your users might be subject to?**

Currently, we are not planning to implement a full Know Your Customer (KYC) process for all users, but we do recognize the importance of complying with relevant regulations and legal frameworks. To ensure compliance and mitigate risks associated with money laundering, fraud, and other financial crimes, we have implemented the following mechanisms:

- **Decentralized Compliance Tools:** We will integrate decentralized compliance tools to verify the legitimacy of projects and tokens listed on the platform. This includes conducting smart contract audits, checking for compliance with financial regulations, and restricting users from countries that our platform may violate their own laws / regulations (subject to Defi), and prevent users from countries that are on international sanction list by UN, by blocking IP addresses.

Note: It is a subject we have thought about it before you ask. And we intend to craft a well written platform policy by hiring a good lawyer accordingly. Which we have mentioned in the office hour discussion with Keegan. And if necessary we will have an option for our users to do KYC from well established third-party KYC organization.

By combining risk-based approaches, compliance with local laws, and the use of decentralized verification tools, we are confident in our ability to meet the regulatory standards necessary while fostering a trustworthy **Decentralized environment** for users.

**Your comments on the advantages of ink! and Polkadot SDK are a little vague. Could you provide an architecture diagram or sequence diagrams for how you imagine - especially cross-chain transactions - would work?**

- Our choice of ink! (a smart contract language for Substrate-based blockchains) and Polkadot SDK stems from the need to build a highly scalable, secure, and interoperable presale launchpad and trading platform. Here are some of the key advantages that ink! and the Polkadot SDK **we can use**. ( **Please note** we have not yet built the Polkadot Substate based contracts yet ). Our grants application includes the development of contracts. )

Just providing you with our plans and feasibilities below:

- **Ink! for Smart Contracts:** Ink! is designed to provide a secure, efficient, and developer-friendly environment for building smart contracts on the Polkadot ecosystem. It is a Rust-based language that compiles to WebAssembly (Wasm), allowing for highly performant contracts with minimal gas costs. Ink! also integrates seamlessly with Substrate, enabling us to build modular and customizable blockchain-based applications while leveraging Substrate’s security guarantees.

- **Polkadot SDK for Cross-Chain Interoperability:** The Polkadot SDK provides the tools necessary to facilitate interoperability across various parachains, making it an ideal platform for our project. By utilizing Polkadot's shared security model, we can easily deploy our presale launchpad across multiple parachains, ensuring scalability while maintaining the security and decentralization of the platform. Simply to put it, we intend to deploy the same or similar presale contract to Acala & Kusama.
But this grants covers only infra & contract built on Polkadot Relay chain. 

We are not building DEX at this point so users are not exchanging or swapping token from one chain to other. Could be a good idea though for the future development.

**For Parachains implementations, we envision the followings:.**

Token Issuance + Presale Deployment: User can do token issuance + Presale Deployment on Polkadot &  parachain (e.g., Acala, Kusama ), 

**Concept**
+----------------------+        +-----------------------+        +-------------------+
|  User Interface      | <----> |  Polkadot Relay Chain  | <----> |  Parachain A (Token Presale) |
+----------------------+        +-----------------------+        +-------------------+
       |                                 |                                  |
       v                                 v                                  v
+----------------+           +-------------------+            +-------------------+
| Token Transfer | <-------> | Cross-Chain Msg.   | <-------> | Ink! Smart Contract|
+----------------+           |   (XCMP)           |            +-------------------+
                             +-------------------+

Perhaps we can remove parachain implementations from the grant application and concentrate only on Polkadot relay chain only for this grants. And revisit parachain implementations at a later stages. 

**Polkadot Relay Chain:** As the core chain in the Polkadot network, the Relay Chain will serve as the backbone for communication across all parachains and will facilitate the core functionality of our presale platform.

Kusama Parachain: Kusama provides an ideal testbed for our platform. By deploying on Kusama, we can quickly iterate and test features in a live environment before deploying to Polkadot.

Our team members and developers are not same. As some devs are hired to do the job needed.
Our devs github accounts:

https://github.com/mjkid221 - ( Dev )
https://github.com/SunlightLuck - ( Dev )
https://github.com/chainemesis - ( Dev & Team member )

We will update the application again and only make it for Polkadot relay chain implementation/integration. And I agree 5 weeks is too short to develop the contracts. But we are too excited and willing to do our best and get this live as quick as possible to catch the bull market :) and develop further. Please feel free to ask any more clarifications you need. And we are happy to answer. Thanks.

**Comment by chainemesis** [comment]
I have read and hereby sign the Contributor License Agreement.

**Comment by keeganquigley** [comment]
Thanks for the application @hydrapad and for your thorough answers. To be honest, I don't understand your above reply. A lot of it doesn't make sense. For one, the relay chain doesn't run smart contracts and doesn't plan to. Also, I'm not sure why a full parachain is needed for token presales. Additionally, many of your above explanations seem as though they are directed at users unfamiliar with the Polkadot ecosystem.

In the call we discussed the possibility of developing a platform on Asset Hub using Solidity contracts, none of which is present here. After further investigation, I'm not convinced that your team has addressed all of the legal complications that come along with running a platform such as this, such as protection against rug pulls. Therefore I think there is too much risk to fund something like this. So I won't vote to approve it, but I wish you the best of luck in securing funding for the platform.

**Comment by hydrapad** [comment]
No problem, we will close the application. Thanks.
